### PR TITLE
Support for attaching multiple networks to a vApp 

### LINKED ISSUE #12959: Adding multi-AZ capability in KKP for VMware Cloud Director
### Info
We already did this for Azure and OpenStack, and this is already working in machine-controller.

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
**What this PR does / why we need it**:
This PR adds support for attaching multiple networks to vApps. This is required for HA setups that require/support multiple availability zones.

**Which issue(s) this PR fixes**:

Fixes #12959 and towards #12650

**What type of PR is this?**

/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
* VMware Cloud Director: Support for attaching multiple networks to a vApp
* [Action Required] The field `ovdcNetwork` in `cluster` and `preset` CRDs is considered deprecated for VMware Cloud Director and `ovdcNetworks` should be used instead.
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: cmd/conformance-tester/pkg/types/secrets.go [hunk 1]
```diff
@@ -106,6 +106,7 @@ type Secrets struct {
 		Organization  string
 		VDC           string
 		OVDCNetwork   string
+		OVDCNetworks  []string
```
Comment (xrstf) [inline cmd/conformance-tester/pkg/types/secrets.go:109]: I don't think we should keep both the old and the new syntax in the conformance-tester's CLI flags. There is no BC guarantee with the tester anyway. We should just have 1 flag, preferably the new one.

Also, this new flag is never set by our code, so the new code isn't tested. Please adjust the hack/ci/run-conformance-tests.sh accordingly.

File: pkg/machine/provider/vmware_cloud_director.go
Comment (xrstf) [inline pkg/machine/provider/vmware_cloud_director.go:1]: Unrelated, but in line 106 can you change the "AWS" to "VCD"?

File: pkg/provider/cloud/vmwareclouddirector/provider.go [hunk 2]
```diff
@@ -160,6 +160,13 @@ func (p *Provider) ValidateCloudSpecUpdate(_ context.Context, oldSpec kubermatic
 	if oldSpec.VMwareCloudDirector.OVDCNetwork != newSpec.VMwareCloudDirector.OVDCNetwork {
 		return fmt.Errorf("updating VMware Cloud Director OVDCNetwork is not supported (was %s, updated to %s)", oldSpec.VMwareCloudDirector.OVDCNetwork, newSpec.VMwareCloudDirector.OVDCNetwork)
 	}
+
+	if oldSpec.VMwareCloudDirector.OVDCNetworks != nil && newSpec.VMwareCloudDirector.OVDCNetworks != nil {
+		if len(oldSpec.VMwareCloudDirector.OVDCNetworks) != len(newSpec.VMwareCloudDirector.OVDCNetworks) {
```
Comment (xrstf) [inline pkg/provider/cloud/vmwareclouddirector/provider.go]: Why is this only checking the length? Is changing [a,b] to [c,d] valid? Maybe using a string set's Equal() function would work better here, if not even using a semantic equality. But I'd probably just use a set.

**Review by xrstf — CHANGES_REQUESTED** [review]
QQ

**Comment by ahmedwaleedmalik** [comment]
Feedback handled @xrstf

**Review by xrstf — APPROVED** [review]
/approve

**Pull request merged**